internal/model/manifest.go

```diff
@@ -177,7 +177,23 @@ func (m Manifest) Validate() error {
 }
 
 func (m1 Manifest) Equal(m2 Manifest) bool {
-	primitivesMatch := m1.Name == m2.Name
+	primitivesEq, dockerEq, k8sEq, dcEq := m1.fieldGroupsEqual(m2)
+	return primitivesEq && dockerEq && k8sEq && dcEq
+}
+
+// ChangesInvalidateBuild checks whether the changes from old => new manifest
+// invalidate our build of the old one; i.e. if we're replacing `old` with `new`,
+// should we perform a full rebuild?
+func ChangesInvalidateBuild(old, new Manifest) bool {
```

Comment:
Is this change necessary to fix the bug in ch2703? I was under the impression that it was just the change to the manifest equality function.

Reply:
I meannnn strictly speaking it's not necessary for just the bug fix -- however, without this code, every change to TriggerMode also causes the manifest to rebuild, which is unnecessary and so makes it a pain to change TriggerMode.

Reply:
If a resource goes from manual to auto, shouldn't it trigger a rebuild since there might be pending changes?

Or is that handled by the build controller?

Reply:
Good question! I believe it's handled by the build controller but i'll write a test

Reply:
Test added -- good catch @jazzdan! (it IS indeed handled by the build controller.)